### PR TITLE
handle client-side history changes

### PR DESCRIPTION
**Pull request by rmorshea**
closes: #16 

Also adds a `*` pattern to the simple router that matches anything (used as catch-all route).

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -14,13 +20,13 @@ jobs:
...
       - name: Run Tests
-        run: nox -s test
+        run: nox -t test
 
   environments:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
```
Comment (Archmonger) [inline .github/workflows/test.yaml:29]: Do we want to remove 3.7 and 3.8?

File: reactpy_router/core.py [hunk 2]
```diff
@@ -113,9 +117,9 @@ def _match_route(
     return None
 
 
-_link = export(
+_link, _history = export(
```
Comment (Archmonger) [inline reactpy_router/core.py:120]: Need to create an issue detailing our long term solution for back-button behavior

**Review by Archmonger — APPROVED** [review]

**Comment by Archmonger** [comment]
We might want to get this merged soon. People other than me are starting to use `reactpy-router`.

File: reactpy_router/core.py (hunk 2 above)
Comment (rmorshea) [reply reactpy_router/core.py:126]: That's here: https://github.com/reactive-python/reactpy-router/issues/19

**Pull request merged**